src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -961,13 +961,13 @@ private static bool NumberBufferToDouble(ref NumberBuffer number, ref double val
         private static class DoubleHelper
         {
             public static unsafe uint Exponent(double d) =>
-                (*((uint*)&d + 1) >> 20) & 0x000007ff;
+                (*((uint*)&d + (BitConverter.IsLittleEndian ? 1 : 0)) >> 20) & 0x000007ff;
 
             public static unsafe ulong Mantissa(double d) =>
                 *((ulong*)&d) & 0x000fffffffffffff;
 
             public static unsafe bool Sign(double d) =>
-                (*((uint*)&d + 1) >> 31) != 0;
```

Comment:
This should be rather written as `bool Sign(double d) => *(long*)&d < 0`. It should produce better native code than the current version *and* it will be big-endian friendly by construction.

Reply:
ASM code for completeness:
```asm
       48837C240800         cmp      qword ptr [rsp+08H], 0
       0F9CC0               setl     al
       0FB6C0               movzx    rax, al
```
And fortunately this also works well on 32 bit platforms.
```asm
G_M55886_IG02:
       837D0C00     cmp      dword ptr [ebp+0CH], 0
       0F9CC0       setl     al
       0FB6C0       movzx    eax, al
```

Reply:
But beware that other 32 bit JITs (e.g. JIT32) will generate rather poor code:
```asm
015E047B 83 7D 0C 00          cmp         dword ptr [ebp+0Ch],0  
015E047F 7F 08                jg          015E0489  
015E0481 7C 0A                jl          015E048D  
015E0483 83 7D 08 00          cmp         dword ptr [ebp+8],0  
015E0487 72 04                jb          015E048D  
015E0489 33 C0                xor         eax,eax  
015E048B EB 05                jmp         015E0492  
015E048D B8 01 00 00 00       mov         eax,1  
```